spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -305,8 +302,11 @@ protected Object handleRequestMessage(Message<?> requestMessage) {
 						(StringUtils.hasText(this.temporaryFileSuffix) && tempFile.exists()));
 
 		if (!ignore) {
-
 			try {
+				if (!resultFile.exists()) {
+					tempFile.getParentFile().mkdirs();
+					resultFile.getParentFile().mkdirs();
```

Comment:
I think, for safety, we should only call `mkdirs` if there's a file separator in the path. Otherwise, this would be a behavior change for simple filenames. See the javadocs for mkdirs...

     * @throws  SecurityException
     *          If a security manager exists and its <code>{@link
     *          java.lang.SecurityManager#checkRead(java.lang.String)}</code>
     *          method does not permit verification of the existence of the
     *          named directory and all necessary parent directories; or if
     *          the <code>{@link
     *          java.lang.SecurityManager#checkWrite(java.lang.String)}</code>
     *          method does not permit the named directory and all necessary
     *          parent directories to be created